### PR TITLE
Added Block Sort Algorithm

### PR DESCRIPTION
**Pull request by SportingDot1024**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Comment by SportingDot1024** [comment]
Would anyone be able to review this pull request?

File: sorts/block_sort.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        List[int]: The sorted list.
+
+    Examples:
+        >>> block_sort([3, 1, 4, 1, 5, 9, 2, 6, 5, 3, 5])
+        [1, 1, 2, 3, 3, 4, 5, 5, 5, 6, 9]
+        >>> block_sort([5, 4, 3, 2, 1])
+        [1, 2, 3, 4, 5]
+        >>> block_sort([])
+        []
+    """
```
Comment (AasheeshLikePanner) [inline sorts/block_sort.py:22]: Can you add edge cases and some larger input lists like this:

```suggestion
    """
    >>> block_sort([1, 1, 1, 1, 1])
        [1, 1, 1, 1, 1]
        >>> block_sort([7, 6, 5, 4, 3, 2, 1])
        [1, 2, 3, 4, 5, 6, 7]
        >>> block_sort([10, 9, 8, 7, 6, 5, 4, 3, 2, 1])
        [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
        >>> block_sort([1])
        [1]
        >>> block_sort([2, 1])
        [1, 2]
        >>> block_sort([10, 20, 30, 40, 50, 60, 70, 80, 90, 100])
        [10, 20, 30, 40, 50, 60, 70, 80, 90, 100]
        >>> block_sort([9, 8, 7, 6, 5, 4, 3, 2, 1, 0])
        [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
        >>> block_sort([1000, 100, 10, 1, 10000, 100000, 1000000])
        [1, 10, 100, 1000, 10000, 100000, 1000000]
```

Comment (SportingDot1024) [reply sorts/block_sort.py:22]: ok

Comment (SportingDot1024) [reply sorts/block_sort.py:38]: I have added more testcases

**Pull request closed without merging**